ang/taskRunner.js

```diff
@@ -20,138 +20,146 @@
-          $scope.task = Object.assign({}, result.values);
-          $scope.isTaskSuccessfullyLoaded = true;
+      }).then((result) => {
+        if (result.is_error) {
+          throw new Error(result.error_message);
         } else {
-          $scope.showError('Getting task api returns error: ' + result.error_message);
+          $scope.state = "loaded";
+          $scope.task = structuredClone(result.values);
+          $scope.inputRequired = [1, "1"].includes($scope.task["input_required"]);
         }
-
-        $scope.isTaskLoading = false;
-        $scope.$apply();
+      }).catch((error) => {
+        CRM.alert("Failed to load task", "Error", "error");
+        $scope.state = "error";
+        $scope.errors.push(error);
       });
     }
 
-    $scope.runTask = async function() {
-      $scope.cleanErrors();
+    $scope.runTask = async () => {
+      $scope.errors = [];
+      $scope.logs = [];
+      $scope.state = "running";
 
-      if ($scope.isInputValueRequired()) {
-        if ($scope.inputValue.length === 0) {
-          $scope.showError('Input value is required');
-        }
+      if ($scope.inputRequired && $scope.inputValue.length < 1) {
+        $scope.state = "error";
+        $scope.errors.push(new Error("Input value is required"));
+        // $scope.$apply();
+        return;
       }
 
-      if ($scope.errors.length > 0 ) return;
-
-      const data = {
+      const taskExecResult = await $scope.api3("Sqltask", "execute", {
+        async: $scope.backgroundQueueEnabled,
+        input_val: $scope.inputValue,
         task_id: taskId,
-        async: await $scope.isBackgroundQueueEnabled(),
-      };
-
-      if ($scope.isInputValueRequired()) {
-        data['input_val'] = $scope.inputValue;
-      }
-
-      $scope.isShowLogs = true;
-      $scope.isTaskRunning = true;
-      $scope.resultLogs = [];
-      $scope.$apply();
-
-      const taskResult = await new Promise((resolve, reject) => {
-        CRM.api3("Sqltask", "execute", data).done(resolve).fail(reject);
-      }).catch($scope.handleExecutionError);
-
-      if (taskResult.is_error !== 0) {
-        $scope.handleExecutionError(new Error(taskResult.error_message));
-      }
-
-      if (!data.async) {
-        CRM.alert("Task execution completed", "Task execution", 'success');
+      }).then((result) => {
+        if (result.is_error) throw new Error(result.error_message);
+        return result;
+      }).catch((error) => {
+        console.error(error);
+        $scope.state = "error";
+        $scope.errors.push(error);
+      });
 
-        $scope.isTaskRunning = false;
-        $scope.resultLogs = taskResult.values.logs;
+      if ($scope.state === "error") {
+        CRM.alert("Task execution failed", "Task execution", "error");
+        $scope.runButtonLabel = "Run again";
+        $scope.$apply();
+        return;
+      } else if (!$scope.backgroundQueueEnabled){
+        CRM.alert("Task execution completed", "Task execution", "success");
```

Comment:
Minor issue: The message here (i.e. in scenarios where background queue is disabled) should also depend on `error_count`.